Lib/test/test_socket.py

```diff
@@ -3888,29 +3940,53 @@ def checkTruncatedArray(self, ancbuf, maxdata, mindata=0):
         self.checkFDs(fds)
 
     def testCmsgTruncLen0(self):
+        if sys.platform == "darwin" and hunting_for_refleaks():
+            return
+
         self.checkTruncatedArray(ancbuf=socket.CMSG_LEN(0), maxdata=0)
 
     def _testCmsgTruncLen0(self):
+        if sys.platform == "darwin" and hunting_for_refleaks():
+            return
+
         self.createAndSendFDs(1)
 
     def testCmsgTruncLen0Plus1(self):
+        if sys.platform == "darwin" and hunting_for_refleaks():
+            return
+
         self.checkTruncatedArray(ancbuf=socket.CMSG_LEN(0) + 1, maxdata=1)
 
     def _testCmsgTruncLen0Plus1(self):
+        if sys.platform == "darwin" and hunting_for_refleaks():
+            return
+
         self.createAndSendFDs(2)
 
```

Comment:
Using decorator might be better way if we introduce new utility function
```suggestion
@unittest.skipIf(hunting_for_refleaks() and sys.platform == 'darwin',
                 'See: gh-80931')
```

Reply:
We cannot use `skipIf` here because AFAIK there is no way to detect a ref leak hunting run while loading tests.

I did switch to some custom decorators to clean up the code, the body of the test functions once again contain only test related code and not bookkeeping like this.